app/components/badge-github-actions.js

```diff
@@ -0,0 +1,56 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { alias } from '@ember/object/computed';
+
+export default Component.extend({
+  tagName: 'span',
+  repository: alias('badge.attributes.repository'),
+  imageUrl: computed('badge.attributes.{repository,workflow_enc,branch,event}', function() {
+    const url = new URL(`https://github.com/${this.repository}/workflows/${this.workflow_enc}/badge.svg`);
+
+    if (this.branch !== '') {
+      url.searchParams.set('branch', this.branch);
+    }
+
+    if (this.event !== '') {
+      url.searchParams.set('event', this.event);
+    }
+
+    return url.href;
+  }),
+  url: computed('badge.attributes.{repository,workflow_enc,branch,event}', function() {
+    const url = new URL(`https://github.com/${this.repository}/actions`);
+
+    if (this.workflow_enc !== '') {
+      url.searchParams.set('workflow', this.workflow_enc);
+    }
+
+    if (this.branch !== '') {
+      url.searchParams.set('branch', this.branch);
+    }
+
+    if (this.event !== '') {
+      url.searchParams.set('event', this.event);
+    }
+
+    return url.href;
+  }),
+  workflow: computed('badge.attributes.workflow', function() {
+    return this.get('badge.attributes.workflow');
+  }),
+  workflow_enc: computed('badge.attributes.workflow', function() {
+    return this.get('badge.attributes.workflow')
+      .split('/')
+      .map(encodeURIComponent)
+      .join('/');
+  }),
+  branch: computed('badge.attributes.branch', function() {
+    return encodeURIComponent(this.get('badge.attributes.branch') || 'master');
```

Comment:
```suggestion
    return this.get('badge.attributes.branch') || 'master';
```

Reply:
I think we want to encode branch names, for branches with `/` in them.

Reply:
Ah wait, `/` works in queries. And we also don't want to display the branch name encoded in the (alt-)text.

Reply:
Branch names do need to be encoded, but `URLSearchParams` takes care of that. (It’s the name that needs the explicit `encodeURIComponent` because it’s a path component instead of a part of the params.)